test/11_array.jl

```diff
@@ -522,6 +522,56 @@ for isordered in (false, true)
             @test x[1] === x.pool.valindex[3]
             @test x[2] === x.pool.valindex[1]
             @test levels(x) == ["c", "a", "b"]
+
+            a1 = 3:200
+            a2 = 300:-1:100
+            ca1 = CategoricalArray(a1)
+            ca2 = CategoricalArray(a2)
+            cca1 = compact(ca1)
+            cca2 = compact(ca2)
+            r = vcat(cca1, cca2)
+            @test r == vcat(a1, a2)
```

Comment:
Ah, and you need to test `levels` and `ordered` after each `vcat` here and below.